src/Raven.Client/Documents/Session/AbstractDocumentQuery.cs

```diff
@@ -1798,6 +1798,12 @@ private object TransformValue(WhereParams whereParams, bool forRange = false)
                     ? ((DateTimeOffset)whereParams.Value).ToString(DefaultFormat.DateTimeOffsetFormatsToWrite, CultureInfo.InvariantCulture)
                     : whereParams.Value;
             }
+#if FEATURE_DATEONLY_TIMEONLY_SUPPORT
+            if (type == typeof(DateOnly))
+                return ((DateOnly)whereParams.Value).ToString(DefaultFormat.TimeOnlyAndDateOnlyFormatToWrite, CultureInfo.InvariantCulture);
```

Comment:
Having single field  - `DefaultFormat.TimeOnlyAndDateOnlyFormatToWrite` for both types looks a bit fragile. I know that effectively it's the same but I think that separation would be better. Those are very different types. Same as you have different converters.